src/main/java/com/avides/spring/rabbit/listener/AbstractSpringRabbitListener.java

```diff
@@ -0,0 +1,83 @@
+package com.avides.spring.rabbit.listener;
+
+import org.springframework.amqp.core.MessageProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.util.StringUtils;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+
+/**
+ * Abstract implementation of {@link SpringRabbitListener} with metrics.
+ * <p>
+ * Implementing classes shall either override {@link #handleEvent(Object)} or {@link #handleEvent(Object, MessageProperties)}.
+ *
+ * @param <T> expected type of the incoming object
+ */
+public abstract class AbstractSpringRabbitListener<T> implements SpringRabbitListener<T>
```

Comment:
Due to the fact that this class has a strong relation to metrics, we could consider renaming this class. This could make this property more obvious.